tests/unit/python/foglamp/services/core/api/test_auth_mandatory.py

```diff
@@ -683,3 +685,65 @@ def auth_token_fixture(self, mocker, is_admin=True):
         patch_refresh_token.assert_called_once_with(ADMIN_USER_HEADER['Authorization'])
         patch_validate_token.assert_called_once_with(ADMIN_USER_HEADER['Authorization'])
         patch_logger_info.assert_called_once_with('Received %s request for %s', 'PUT', '/foglamp/admin/2/reset')
+
+    @pytest.mark.parametrize("auth_method, request_data, ret_val", [
+        ("certificate", "-----BEGIN CERTIFICATE----- Test -----END CERTIFICATE-----", (2, "token2", False))
+    ])
+    async def test_login_auth_certificate(self, client, auth_method, request_data, ret_val):
+        TEXT_HEADER = {'content-type': 'text/plain'}
+
+        async def async_mock():
+            return ret_val
+
+        async def async_get_user():
+            return {'role_id': '2', 'id': '2', 'uname': 'user'}
+
+        with patch.object(middleware._logger, 'info') as patch_logger_info:
+            with patch.object(server.Server, "auth_method", auth_method) as patch_auth_method:
+                with patch.object(SSLVerifier, 'get_subject', return_value={"commonName": "user"}) as patch_common_name:
+                    with patch.object(User.Objects, 'verify_certificate', return_value=asyncio.sleep(.1)) as patch_verify_certificate:
+                        with patch.object(User.Objects, 'certificate_login',
+                                          return_value=async_mock()) as patch_user_login:
+                            with patch.object(User.Objects, 'get', return_value=async_get_user()) as patch_get_user:
+                                with patch.object(auth._logger, 'info') as patch_logger:
+                                    req_data = request_data
+                                    resp = await client.post('/foglamp/login', data=req_data, headers=TEXT_HEADER)
+                                    assert 200 == resp.status
+                                    r = await resp.text()
+                                    actual = json.loads(r)
+                                    assert ret_val[0] == actual['uid']
+                                    assert ret_val[1] == actual['token']
+                                    assert ret_val[2] == actual['admin']
```

Comment:
Please add patch assertions for `patch_logger_info, patch_auth_method, patch_common_name, patch_verify_certificate, patch_user_login, patch_get_user, patch_logger`

Reply:
removed